triton/stream_reader_test.go

```diff
@@ -80,3 +80,87 @@ func TestNewStreamReader(t *testing.T) {
 
 	sr.Stop()
 }
+
+func TestProcessStreamToChanErrors(t *testing.T) {
+	// This tests whether the stream can recover from and ignore bad data
+	svc := newTestKinesisService()
+	st := newTestKinesisStream("test-stream")
+	s1 := newTestKinesisShard()
+
+	r1 := make(map[string]interface{})
+	r1["value"] = "a"
+	s1.AddOverlengthRecord(SequenceNumber("a"), r1)
+	r2 := make(map[string]interface{})
+	r2["value"] = "b"
+	s1.AddRecord(SequenceNumber("b"), r2)
+	st.AddShard(ShardID("0"), s1)
+
+	s2 := newTestKinesisShard()
+	s2.AddBadEncodingRecord(SequenceNumber("c"))
+	r3 := make(map[string]interface{})
+	r3["value"] = "d"
+	s2.AddRecord(SequenceNumber("d"), r3)
+	st.AddShard(ShardID("1"), s2)
+
+	svc.AddStream(st)
+
+	db := openTestDB()
```

Comment:
I don't think you should do any checkpointing here. Either remove all the checkpointer info or add an issue to tackle this later.